crates/sui-indexer/src/indexer.rs

```diff
@@ -73,13 +73,14 @@ impl Indexer {
         spawn_monitored_task!(fetcher.run());
 
         let objects_snapshot_processor = ObjectsSnapshotProcessor::new_with_config(
+            rest_client.clone(),
             store.clone(),
             metrics.clone(),
             snapshot_config,
         );
         spawn_monitored_task!(objects_snapshot_processor.start());
 
-        let checkpoint_handler = new_handlers(store, metrics.clone()).await?;
+        let checkpoint_handler = new_handlers(store, rest_client.clone(), metrics.clone()).await?;
```

Comment:
rest client is necessary to tell the gap between latest committed cp and the highest known cp on FN.